python/tests/test_logistic_regression.py

```diff
@@ -2092,3 +2108,110 @@ def test_quick_double_precision(
         reg_factors=reg_factors,
         float32_inputs=float32_inputs,
     )
+
+
+@pytest.mark.slow
+def test_sparse_int64() -> None:
+    gpu_number = 1
+    cpu_number = 32
+    data_shape = (int(1e7), 2200)
+    fraction_sampled_for_test = (
+        1.0 if data_shape[0] <= 100000 else 100000 / data_shape[0]
+    )
+    n_classes = 8
+    tolerance = 0.001
+    est_params: Dict[str, Any] = {
+        "regParam": 0.02,
+        "maxIter": 10,
+        "standardization": False,  # reduce GPU memory since standardization copies the value array
+    }
+    density = 0.1
+
+    data_gen_args = [
+        "--n_informative",
+        f"{math.ceil(data_shape[1] / 3)}",
+        "--num_rows",
+        str(data_shape[0]),
+        "--num_cols",
+        str(data_shape[1]),
+        "--output_num_files",
+        f"{cpu_number}",
+        "--dtype",
+        "float64",
+        "--feature_type",
+        "vector",
+        "--output_dir",
+        "./temp",
+        "--n_classes",
+        str(n_classes),
+        "--random_state",
+        "0",
+        "--logistic_regression",
+        "True",
+        "--density",
+        str(density),
+        "--use_gpu",
+        "True",
+    ]
+
+    from . import conftest
+
+    conftest._spark.stop()
+
+    from pyspark.sql import SparkSession
+
+    builder = SparkSession.builder.appName(name="spark-rapids-ml with large dataset")
+
+    spark_conf = {
+        "spark.master": f"local[{cpu_number}]",  # avoid local[1] because driver will throw out a java.lang.OutOfMemoryError "Required array length is too large"
+        "spark.rapids.ml.uvm.enabled": True,
+    }
+
+    for key, value in spark_conf.items():
+        builder.config(key, value)
+
+    conftest._spark = builder.getOrCreate()
+
+    import importlib
+
+    from . import sparksession
+
+    importlib.reload(sparksession)
+
+    data_gen = SparseRegressionDataGen(data_gen_args)
+    df, _, _ = data_gen.gen_dataframe_and_meta(conftest._spark)
+    df = df.cache()
+
+    # check nnz > int32.max
+    def get_nnz(sparse_vec: SparseVector) -> int:
+        return len(sparse_vec.values)
+
+    udf_get_nnz = udf(get_nnz, LongType())
+    nnz_df = df.select(udf_get_nnz("features").alias("nnz"))
```

Comment:
Is this slow?  udf is invoked on each row I believe.

Reply:
Good catch. I replaced it with mapInPandas. Runtime was reduced by 45 seconds.